File: edukate-common/src/main/java/io/github/sanyavertolet/edukate/common/services/Notifier.java
package io.github.sanyavertolet.edukate.common.services;

import io.github.sanyavertolet.edukate.common.notifications.BaseNotificationCreateRequest;
import reactor.core.publisher.Mono;

public interface Notifier {
    Mono<String> notify(BaseNotificationCreateRequest notificationCreationRequest);
}

File: edukate-common/src/main/kotlin/io/github/sanyavertolet/edukate/common/services/Notifier.kt
package io.github.sanyavertolet.edukate.common.services

import io.github.sanyavertolet.edukate.common.notifications.BaseNotificationCreateRequest
import reactor.core.publisher.Mono

interface Notifier {
    fun notify(notificationCreationRequest: BaseNotificationCreateRequest): Mono<String>
}
